### PR TITLE
:sparkles: Changes File class to allow for certain endpoints more easily

### PR DESCRIPTION
**Pull request by Lunarmagpie**
### Changes

-   `adds`: File can now be converted into an URI with the `get_uri()` method. Aiohttp Payloads are now constructed with `create_form` function in the File file.
-   `fixed`: `Guild.create_emoji()` now takes a File object as an input.
-   `improvements`: ...

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

File: pincer/objects/message/file.py [hunk 1]
```diff
@@ -28,8 +31,58 @@
...
+
+    Returns
+    -------
+    Optional[str]
+        The file extension or :data:`None`
+    """
+    path = os.path.splitext(filename)
+    if len(path) >= 2:
+        return path[1][1:]
+    return None
```
Comment (Sigmanificient) [inline pincer/objects/message/file.py:81]: `return None` is uneeded

File: pincer/objects/message/file.py [hunk 2]
```diff
@@ -119,5 +179,24 @@ def from_pillow_image(
 
         return cls(
             content=img_bytes,
+            image_format=image_format,
             filename=filename
         )
+
+    def get_uri(self) -> str:
```
Comment (Sigmanificient) [inline pincer/objects/message/file.py:186]: property?

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pincer/objects/message/file.py (hunk 1 above)
Comment (Lunarmagpie) [reply pincer/objects/message/file.py:81]: You have to explicitly state `return None` if there are any returns according to pep8
> Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should. If any return statement returns an expression, any return statements where no value is returned should explicitly state this as return None, and an explicit return statement should be present at the end of the function (if reachable)

https://www.python.org/dev/peps/pep-0008/

File: pincer/objects/message/file.py (hunk 2 above)
Comment (Lunarmagpie) [reply pincer/objects/message/file.py:186]: good idea

File: pincer/objects/message/file.py [hunk 3]
```diff
@@ -106,7 +166,7 @@ def from_pillow_image(
             )
 
         if image_format is None:
-            image_format = os.path.splitext(filename)[1][1:]
+            image_format = _get_file_extension(filename)
 
             if image_format == "jpg":
                 image_format = "jpeg"
```
Comment (trag1c) [inline pincer/objects/message/file.py:172]: Shouldn't the 2nd if statement be unindented?

File: pincer/objects/guild/guild.py [hunk 4]
```diff
@@ -1295,8 +1298,8 @@ async def create_emoji(
         self,
         *,
         name: str,
-        image: str,
-        roles: List[Snowflake],
+        image: File,
+        roles: List[Snowflake] = [],
```
Comment (trag1c) [inline pincer/objects/guild/guild.py:1302]: Was this intended?

File: pincer/objects/message/file.py [hunk 5]
```diff
@@ -3,21 +3,24 @@
...
 from __future__ import annotations
 
+from base64 import b64encode
 import os
 from dataclasses import dataclass
 from io import BytesIO
-from typing import Any
-from typing import TYPE_CHECKING
+from json import dumps
+from typing import TYPE_CHECKING, Tuple
```
Comment (trag1c) [inline pincer/objects/message/file.py:11]: `Tuple` should be moved to L18, no?

File: pincer/objects/guild/guild.py (hunk 4 above)
Comment (Lunarmagpie) [reply pincer/objects/guild/guild.py:1302]: Yeah, having to manually specify an empty arrow for no permissions didn't make sense imo.

File: pincer/objects/message/file.py (hunk 3 above)
Comment (Lunarmagpie) [reply pincer/objects/message/file.py:172]: good call

File: pincer/objects/message/file.py (hunk 5 above)
Comment (Lunarmagpie) [reply pincer/objects/message/file.py:11]: yeah

**Review by Sigmanificient — APPROVED** [review]

File: pincer/exceptions.py [hunk 6]
```diff
@@ -163,6 +163,10 @@ class InvalidUrlError(PincerError, ValueError):
     """Exception raised when an invalid url has been provided."""
 
 
+class ImageEncodingError(PincerError):
+    """Exception raised when an image cannot be encoded for Discord"""
+
+
```
Comment (Endercheif) [inline pincer/exceptions.py:169]: Add to `__init__.py`

File: pincer/objects/message/file.py [hunk 7]
```diff
@@ -28,8 +31,58 @@
...
+    """
+    Returns the file extension from a str if it exists, otherwise
+    return :data:`None`.
+
+    filename : str
+        The filename
+
+    Returns
+    -------
+    Optional[str]
```
Comment (Endercheif) [inline pincer/objects/message/file.py:75]:
```suggestion
    Optional[:class:`str`]
```

**Review by Endercheif — CHANGES_REQUESTED** [review]

**Review by Endercheif — APPROVED** [review]

**Pull request merged**